packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -452,11 +452,10 @@ export class ProfilesCache {
     }
 
     private shouldRemoveTokenFromProfile(profile: zowe.imperative.IProfileLoaded, baseProfile: zowe.imperative.IProfileLoaded): boolean {
-        return (baseProfile?.profile?.host &&
-            baseProfile?.profile?.port &&
+        return ((baseProfile?.profile?.host || baseProfile?.profile?.port) &&
```

Comment:
I think adding the check for user and password is good 👍 

Could we restore the old logic for port on lines 455 and 458?

(same comment as on the v2 version of this PR 😋)